cobigen/cobigen-core/src/main/java/com/devonfw/cobigen/impl/util/ConfigurationFinder.java

```diff
@@ -102,12 +102,22 @@ public static URI findTemplatesLocation() {
       // use new template set configuration
       Path templateSetsFolderLocation = getTemplatesFolderLocation(cobigenHome, configFile, templateSetsLocation);
       if (templateSetsFolderLocation != null && Files.exists(templateSetsFolderLocation)) {
+        if (!Files.exists(templateSetsFolderLocation)) {
```

Comment:
This is dead code as you checked beforehand whether `Files.exists(templateSetsFolderLocation)`